ext/uri/uri_parser_rfc3986.c

```diff
@@ -99,6 +99,13 @@ ZEND_ATTRIBUTE_NONNULL static UriUriA *get_uri_for_reading(php_uri_parser_rfc398
 	}
 }
 
+ZEND_ATTRIBUTE_NONNULL static UriUriA *get_uri_for_writing(uri_internal_t *internal_uri)
+{
+	php_uri_parser_rfc3986_uris *uriparser_uris = internal_uri->uri;
+
+	return &uriparser_uris->uri;
+}
```

Comment:
From what I see, the implementation now has exactly the bug that #19588 was supposed to avoid: You are not resetting the normalized URI when writing and during cloning the normalized URI is cloned.

Consider this test case:

```php
<?php

$uri = \Uri\Rfc3986\Uri::parse('https://example.com/path-%65xample/');
var_dump($uri->getPath());
var_dump($uri->getRawPath());

$newUri = $uri->withPath('/updated-path');
var_dump($newUri->getPath());
var_dump($newUri->getRawPath());
```